### DISCUSSION TITLE
debugging modules with nuget references

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#727] [General] [answered]
I've created a module with a server nuget reference to RepoDB.  To deploy it to the framework, so I've included the file in the nuget package.  To debug it, I feel I should copy the required files to the framework folder via debug.cmd  I get stuck at this point because there are no referenced DLLs in my module.  How can I debug a module with a nuget package dependancy without adding a reference to the framework?  

![image](https://user-images.githubusercontent.com/311063/91918503-ef672880-ec77-11ea-8983-87c9ef91ed7c.png)
![image](https://user-images.githubusercontent.com/311063/91918560-1887b900-ec78-11ea-9622-e40a763e0f77.png)

![image](https://user-images.githubusercontent.com/311063/91918677-774d3280-ec78-11ea-95d6-bac59d5d922f.png)

**Comment by markdav-is** [comment]
the work-around I'm using at the moment is adding those packages to the framework

**Comment by mikecasas** [comment, marked as answer]
@nohorse I'm using Dapper in one of my external modules, which is a similar ORM. Try adding the below to your ```Server.csproj``` file in the ```PropertyGroup``` section:

```
    <!--this will force the compiler to copy all of the DLL dependencies from the Nuget package locations to the project /bin-->
    <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
```
Then in your ```debug.cmd``` in the Package project copy the dlls over to the framework.

```
REM Copy Dapper to the Bin folder (and any dll dependencies)
XCOPY "..\Server\bin\Debug\netcoreapp3.1\Dapper.dll" "..\..\oqtane.framework\Oqtane.Server\bin\Debug\netcoreapp3.1\" /Y
XCOPY "..\Server\bin\Debug\netcoreapp3.1\Dapper.Contrib.dll" "..\..\oqtane.framework\Oqtane.Server\bin\Debug\netcoreapp3.1\" /Y
```


The "trick" is that you may have to copy some of the dependent dlls over too. Some of them may not be initially obvious.

**Reply by markdav-is** [reply]
perfect, this is exactly what I was looking for!
